win32ss/user/ntuser/ime.c

```diff
@@ -80,14 +80,72 @@ NtUserCheckImeHotKey(
     return 0;
 }
 
-
-DWORD
+BOOL
 APIENTRY
 NtUserDisableThreadIme(
-    DWORD dwUnknown1)
+    DWORD dwThreadID)
 {
-    STUB;
-    return 0;
+    PTHREADINFO pti, ptiCurrent;
+    PPROCESSINFO ppi;
+    BOOL ret = FALSE;
+
+    UserEnterExclusive();
+
+    if (!IS_IMM_MODE())
+    {
+        EngSetLastError(ERROR_CALL_NOT_IMPLEMENTED);
+        goto Quit;
+    }
+
+    ptiCurrent = GetW32ThreadInfo();
+
+    if (dwThreadID == (DWORD)-1)
```

Comment:
These seem to be some sort of special IDs. Is there any other code in win32k (or in your already-implemented IME code) that use these? maybe there are some #defines for these?

Reply:
Seems like 0 == just current thread; > 0 == that thread; -1 == all the threads (+ process)

Reply:
> Is there any other code in win32k

Nope.

Reply:
Defined `INVALID_THREAD_ID` and used it.